src/ngraph/op/tensor_iterator.hpp

```diff
@@ -0,0 +1,300 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <vector>
+
+#include "ngraph/lambda.hpp"
+#include "ngraph/op/parameter.hpp"
+#include "ngraph/op/util/fused_op.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        /// \brief  Iterate a body over tensors, accumulating into tensors.
+        class TensorIterator : public util::FusedOp
+        {
+        public:
+            NGRAPH_API
+            static constexpr NodeTypeInfo type_info{"TensorIterator", 0};
+            const NodeTypeInfo& get_type_info() const override { return type_info; }
+            // Forward declarations
+            class SliceInputDescription;
+            class BodyConnectionInputDescription;
+            class ConstantInputDescription;
+
+            TensorIterator() = default;
+            TensorIterator(const OutputVector& values);
+
+            class BodyLambda : public Lambda
```

Comment:
@ayzhuang @diyessi Thanks, it looks suitable for us, but could you explain the reason
Why did you define this class? It looks the same as `Lambda`. 
And the second question is: Why don't you want to use `ngraph::Function` for TensorIterator body?

Reply:
@diyessi thanks for the explanation. We will try to complete the review this week.
The helpers look really useful, thanks a lot!